pom.xml

```diff
@@ -265,15 +265,8 @@
       <id>fatjar</id>
       <modules>
         <module>zookeeper-it</module>
-        <module>zookeeper-contrib/zookeeper-contrib-fatjar</module>
```

Comment:
@ctubbsii 
To build a fatjar, we need to include the `zookeeper-contrib (or zookeeper-contrib/zookeeper-contrib-fatjar)` module either here or under the top-level modules section.

Reply:
Activating the `fatjar` profile will already activate it in the `zookeeper-contrib` parent pom, which activates the the `zookeeper-contrib-fatjar` under that.

Activating it here is redundant, and breaks when you also activate the `full-build` profile.

Removing it here fixes that breakage when both profiles are activated.

To simplify things, I recommend merging the `full-build` and `fatjar` profiles, so that there is a single `full-build` profile that creates the fatjar with all the other contribs. Alternatively, if you want to build the fatjar without the contribs, I recommend moving the fatjar to a module outside of the contribs, so that activation doesn't conflict with the `full-build` profile.

Reply:
(Also, see related discussion on the mailing list regarding moving contribs to separate repos.)

Reply:
> Activating it here is redundant, and breaks when you also activate the full-build profile.

Why one needs to activate both profiles simultaneously?

A separate fat jar profile allows creating a fatjar without having to build zookeeper-client and some other unnecessary modules. 
 

> Alternatively, if you want to build the fatjar without the contribs, I recommend moving the fatjar to a module outside of the contribs, so that activation doesn't conflict with the full-build profile.

This is a good idea.  

@breed  Should we merge zookeeper-contrib-fatjar and the zookeeper-it module?

Reply:
no, i don't think fatjar should be in the zookeeper-it module. while the fatjar does make the tests more convenient, most people using fatjars are not running tests. (they are usually starting the server, or using it as a convenient client.)

Reply:
> i don't think fatjar should be in the zookeeper-it module

I think you may have misunderstood me. I didn't mean to suggest combining the two modules into a single module. What I meant was combining the activation of all the contribs into a single `full-build` profile, and not having a separate fatjar-only profile.

Reply:
> Why one needs to activate both profiles simultaneously?

Because I want to build everything to test the complete build.

Even if you don't want to do this, and really do want these to be mutually exclusive profiles, the build is broken without this change. You need `full-build` to activate the `zookeeper-contrib` parent pom, which is a dependency of the `zookeeper-contrib-fatjar`. But, you can't activate the `fatjar` profile and the `full-build` profile without the changes I made, because together, they try to activate the `zookeeper-contrib-fatjar` module twice, which Maven won't allow.

There are many ways to solve this... I chose one possibility, but there are others.... it depends on what combination of options you want to support.

Personally, I would advise a solution that minimizes the number of profiles: standard build or full build (with contribs). Then, all you need to decide is whether you want the fatjar to be included in the standard build or the `full-build` profile.

Reply:
@sushantmane and @breed : have I answered your questions? Is there more you'd like me to follow up on for this?

Reply:
@ctubbsii 
I think it would be good to move the `zookeeper-contrib-fatjar` module to the top-level directory and have a separate maven profile to build it.

Reply:
> I think it would be good to move the `zookeeper-contrib-fatjar` module to the top-level directory and have a separate maven profile to build it.

I agree it should be moved, but I think it would be better done as its own distinct follow-on task. This PR is already a bit overloaded, don't you think?

Reply:
yes, this patch is almost ready to land